src/Apex-Assignment 2/classes/TrainingSessionManager.cls

```diff
@@ -0,0 +1,150 @@
+/*
+Purpose        : This class has been created to manage Training Session records.
+Created Date   : 11/27/2024
+Created By     : Sarita Rathore
+Revision_Log   : V1.0
+*/
+public with sharing class TrainingSessionManager {
+public enum SessionType { 
+  WEBINAR , WORKSHOP, SEMINAR
+  }
+/*
+ Purpose        : This Method will implement the usecase of Enum SessionType.
+ Return         : void
+ Argument       : SessionType
+ Revision_Log   : V1.0
+ */
+public static void usecaseOfEnum(SessionType type){
+  Training_Session__c newTrainingSession = new Training_Session__c(Name='Training Session-Webinar', Type_Of_Session__c = type.name());
+  try{
+    insert newTrainingSession;
+  }
+  catch(DmlException e){
+    System.debug('Insertion Failed'+ e.getMessage());
+  }
+}
+
+/*
+ Purpose        : This Method will create records in Training Session.
+ Return         : void
+ Argument       : No Argument
+ Revision_Log   : V1.0
+ */
+public static void createRecordsOfTrainingSession(){
+  List<Training_Session__c> newRecords = new List<Training_Session__c>();
+  for(Integer i=0;i<10;i++){
+    newRecords.add(new Training_Session__c(Name = 'TrainingSession'+ i));
+  }
+  try{
+    if(newRecords != Null){
+      insert newRecords;
+    }
+  }
+  catch(DmlException error){
+     System.debug('DML Insertion Failed'+ error.getMessage());
+  }
+  
+}
+/*
+ Purpose        : This Method will read records from Training Session.
+ Return         : void
+ Argument       : Id 
+ Revision_Log   : V1.0
+ */
+public static void readRecordsFromTrainingSession(ID id){
+  List<Training_Session__c> readRecords = new List<Training_Session__c>();
+  for(Training_Session__c trainingSessionsToBeReaded:[SELECT Id,Name FROM Training_Session__c]){
+    if(trainingSessionsToBeReaded.Id == id)
+      readRecords.add(trainingSessionsToBeReaded);
+  }
+  if(!readRecords.isEmpty())
+    System.debug(readRecords);
+  else
+    System.debug('No Such Record is there in Database');
+}
+/*
+ Purpose        : This Method will update records from Training Session.
+ Return         : void
+ Argument       : Id,String
+ Revision_Log   : V1.0
+ */
+public static void updateRecordsFromTrainingSession(ID id, String location){
+  for(Training_Session__c trainingSessionToBeUpdated:[SELECT Id, Name, Location__c FROM Training_Session__c]){
+   try{
+    if(trainingSessionToBeUpdated.Id == id){
+      trainingSessionToBeUpdated.Location__c=location;
+      update trainingSessionToBeUpdated;
```

Comment:
update the query and also we never perform DML inside for loop.

Reply:
Updated the Code, Please Check.